walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,112 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
 
 public final class FindMeetingQuery {
+
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    long meetingDuration = request.getDuration();
+
```

Comment:
It might be worthwhile to add some more edge conditions. E.g. if there are not attendees requested in the meeting or if there are no events ?

Reply:
No events has already been handled. It will return a workable time slot from TimeRange.START_OF_DAY to TimeRange.END_OF_DAY.
For no attendees case, it will not break the code but a test can help reducing the computation.
Thanks for the suggestion.